tiledb/sm/query/sparse_index_reader_base.cc

```diff
@@ -183,6 +190,12 @@ SparseIndexReaderBase::get_coord_tiles_size(
   if (subarray_.is_set())
     tiles_size += fragment_metadata_[f]->cell_num(t) * sizeof(BitmapType);
 
+  if (consolidation_with_timestamps_ && load_timestamps_ &&
+      fragment_metadata_[f]->has_timestamps() && !discard_partial_timestamps) {
```

Comment:
We don't really need to check for consolidation_with_timestamps_... fragment_metadata_[f]->has_timestamps() should already cover that.

For discard_partial_timestamps, let's code it up in conjunction with the fragment timestamps and the array open start/end times.